blog/2025-01-21-Agama-11.mdx

```diff
@@ -0,0 +1,210 @@
+each script would need to be provided as a long string with "\n" marking the end of each line (not
+so nice for a blog post :wink:).
+
+```jsonnet
+{
+  scripts: {
+    pre: [
+      {
+        name: "activate-multipath",
+        body: |||
+          #!/usr/bin/bash
+          systemctl start multipathd.socket multipathd.service
+        |||
+      }
+    ],
+    post: [
+      {
+        name: "enable-sshd",
+        chroot: true,
+        body: |||
+          #!/usr/bin/bash
+          systemctl enable sshd.service
+        |||
+      }
+    ],
+    init: [
+      {
+        name: "run-ansible",
+        url: "https://192.168.1.1/provisioning.sh"
+      }
+    ]
+}
+```
+
+For more details see the [scripts
+section](https://agama-project.github.io/docs/user/unattended/scripts) of the Agama documentation
+site.
+
+## Storage management for unattended installation {#storage}
+
+The Agama configuration format offers a very convenient and powerful approach to configure the
+storage setup of the new system, way more consistent and concise that the corresponding 
+`<partitioning>` section of the AutoYaST profile (which is still fully supported for migration
+purposes).
+
+Agama 11 adds the possibility to define the physical volumes of an LVM volume group by simply
+specifying the disk (or disks) that will be used as a base for the LVM. Agama will take care of
+creating all the needed partitions, honoring any other aspect of the configuration in the process.
+Find a more detailed explanation with examples at the [corresponding
+section](https://agama-project.github.io/docs/user/unattended/storage#generating-physical-volumes)
+of the Agama documentation.
+
+On the other hand, now it is possible to specify TPM-based unlocking of the encrypted devices as
+part of the Agama storage configuration. Thus, users of unattended installation can also deploy
+fully encrypted systems based on TPMv2.
+
+## Automatic generation of documentation and shell completion {#generate}
+
+A comprehensive and up-to-date documentation is key for a project like Agama, especially for users
+of the command-line interface and the HTTP API. And the best way to ensure the documentation is
+always in sync with the current version of Agama is to generate it automagically from the source
+code.
+
+In the case of the CLI, the manual pages, its Markdown variant and also the files needed for
+[shell completion](https://en.wikipedia.org/wiki/Command-line_completion) are all generated from
+sources. You can see the always current result of the Markdown version at the [corresponding
+section](https://agama-project.github.io/docs/user/cli) of the Agama web page.
+
+The HTTP API is also automatically documented via an [OpenAPI](https://www.openapis.org/)
+specification. This will help anyone interested in integrating Agama into any solution or
+infrastructure or even in creating its own client application for Agama.
```

Comment:
It should be probably mentioned that the HTTP API is not stable and is still in development and it might change. Just in case someone would like to write own client...